### ISSUE TITLE
`escape` method for `run` command

### ISSUE THREAD

**Issue by clxmstaab** [deployphp/deployer#2176]
**Is your feature request related to a problem? Please describe.**
we use the `mysql` command line, invoked via deployer `run()` to pass sql data between servers thru deployers ssh tunnel.

the sql command used for the commands can contain any text, e.g. even a `{{abc}}` which deployer tries to parse as a variable.

**Describe the solution you'd like**
I would like some kind of `escape` function which allows me to mask a string which may contain `{{xx}}` and returns me a string which can be passed thru `run()` without beeing interpreted as deployer variables.

in the end my php api looks like 
```
$data = run('mysql --silent --raw --skip-column-names --execute '.AsyncProcess::escapeArgument($query).' '.$dbCredentials->asCliArgs());
```

in the recipe

especially suprising is the error which gets reported, because it mentions a configuration variable not beeing defined and it took me a while to find out that my data contains stuff deployer is trying to interpret

```
In Configuration.php line 89:

  [Deployer\Exception\ConfigurationException]
  Configuration parameter `xx` does not exist.

Exception trace:
  at /home/complexit/.composer/vendor/deployer/deployer/src/Configuration/Configuration.php:89
 Deployer\Configuration\Configuration->get() at /home/complexit/.composer/vendor/deployer/deployer/src/Configuration/Configuration.php:130
 Deployer\Configuration\Configuration->parseCallback() at n/a:n/a
 preg_replace_callback() at /home/complexit/.composer/vendor/deployer/deployer/src/Configuration/Configuration.php:116
 Deployer\Configuration\Configuration->parse() at /home/complexit/.composer/vendor/deployer/deployer/src/functions.php:761
 Deployer\parse() at /home/complexit/.composer/vendor/deployer/deployer/src/functions.php:494
```

**Comment by antonmedv** [comment]
Yes, I got you. Well, some function can be added to functions.php for sure. Would you like to make a PR with it?

**Comment by bsacharski** [comment]
I've implemented in a #2180. 
The `{{` and `}}` are escaped by being replaced with `\{\{` and `\}\}` respectively.
This causes the brackets to not be matched by the regex and as such, they are not being replaced.

Note: the values are always unescaped after being parsed - not sure if that should happen though.

**Comment by antonmedv** [comment]
Closing in favor of #2192
